pkg/kafkaclient/users.go

```diff
@@ -68,9 +68,14 @@ func (k *kafkaClient) ListUserACLs() ([]sarama.ResourceAcls, error) {
 }
 
 // DeleteUserACLs removes all ACLs for a given user
-func (k *kafkaClient) DeleteUserACLs(dn string) (err error) {
+func (k *kafkaClient) DeleteUserACLs(dn string, patternType v1alpha1.KafkaPatternType) (err error) {
+	aclPatternType := AclPatternTypeMapping(patternType)
 	matches, err := k.admin.DeleteACL(sarama.AclFilter{
-		Principal: &dn,
+		Principal:                 &dn,
+		ResourcePatternTypeFilter: aclPatternType,
+		Operation:                 sarama.AclOperationDelete,
+		ResourceType:              sarama.AclResourceTopic,
+		PermissionType:            sarama.AclPermissionAny,
 	}, false)
 	if err != nil {
 		return
```

Comment:
> while going through the PR, I've found this minor bug here

I think its not a bug, it was a "named parameter" as a result it will return the err, but I changed it to more generic way of writing for better readability.